<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>WizardNewContainer</class>
 <widget class="QWizard" name="WizardNewContainer">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>455</width>
    <height>352</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Passwords wizard</string>
  </property>
  <property name="sizeGripEnabled">
   <bool>false</bool>
  </property>
  <property name="wizardStyle">
   <enum>QWizard::AeroStyle</enum>
  </property>
  <property name="options">
   <set>QWizard::CancelButtonOnLeft|QWizard::HelpButtonOnRight|QWizard::NoBackButtonOnStartPage|QWizard::NoDefaultButton</set>
  </property>
  <widget class="QWizardPage" name="wizardPage1">
   <property name="title">
    <string>Container password</string>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="1">
     <widget class="QLineEdit" name="textFileName"/>
    </item>
    <item row="3" column="1">
     <widget class="QLineEdit" name="textPassword">
      <property name="echoMode">
       <enum>QLineEdit::Password</enum>
      </property>
     </widget>
    </item>
    <item row="3" column="0">
     <widget class="QLabel" name="label_2">
      <property name="text">
       <string>Password:</string>
      </property>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QLabel" name="labelFilename">
      <property name="text">
       <string>Filename:</string>
      </property>
     </widget>
    </item>
    <item row="2" column="0" colspan="3">
     <widget class="Line" name="line_2">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item row="4" column="0">
     <widget class="QLabel" name="label_3">
      <property name="text">
       <string>Repeat password:</string>
      </property>
     </widget>
    </item>
    <item row="4" column="1">
     <widget class="QLineEdit" name="textRepeatedPassword">
      <property name="echoMode">
       <enum>QLineEdit::Password</enum>
      </property>
     </widget>
    </item>
    <item row="0" column="2">
     <widget class="QPushButton" name="buttonSelectFile">
      <property name="text">
       <string/>
      </property>
      <property name="icon">
       <iconset resource="../../resources/resource.qrc">
        <normaloff>:/flags_color/icons_black/open_folder2_black.png</normaloff>:/flags_color/icons_black/open_folder2_black.png</iconset>
      </property>
     </widget>
    </item>
    <item row="5" column="1">
     <widget class="QLabel" name="labelPasswordSafety">
      <property name="text">
       <string>Please enter a password!</string>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QLineEdit" name="textOldPassword">
      <property name="echoMode">
       <enum>QLineEdit::Password</enum>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QLabel" name="labelOldPassword">
      <property name="text">
       <string>Old password:</string>
      </property>
     </widget>
    </item>
    <item row="6" column="1">
     <widget class="QLabel" name="labelStatus">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWizardPage" name="wizardPage2">
   <property name="title">
    <string>Password hashing settings</string>
   </property>
   <property name="subTitle">
    <string/>
   </property>
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="0">
     <widget class="QLabel" name="label_4">
      <property name="text">
       <string>Iteration cost:</string>
      </property>
     </widget>
    </item>
    <item row="2" column="0">
     <widget class="QLabel" name="label_6">
      <property name="text">
       <string>Lanes:</string>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QLabel" name="label_5">
      <property name="text">
       <string>Memory cost [kB]:</string>
      </property>
     </widget>
    </item>
    <item row="1" column="1" colspan="2">
     <widget class="QSpinBox" name="spinMemoryCost">
      <property name="whatsThis">
       <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;How many RAM should the algorithm use.&lt;br/&gt;How many free RAM do you have in your system?&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
      <property name="minimum">
       <number>32</number>
      </property>
      <property name="maximum">
       <number>999999999</number>
      </property>
      <property name="singleStep">
       <number>8</number>
      </property>
      <property name="value">
       <number>1024</number>
      </property>
     </widget>
    </item>
    <item row="2" column="1" colspan="2">
     <widget class="QSpinBox" name="spinLanes">
      <property name="whatsThis">
       <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Do you have a multi core processor?&lt;br/&gt;Set this value to the number of cores.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>99999</number>
      </property>
      <property name="value">
       <number>2</number>
      </property>
     </widget>
    </item>
    <item row="0" column="1" colspan="2">
     <widget class="QSpinBox" name="spinIterationCost">
      <property name="whatsThis">
       <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;How many iterations should the algorithm do. &lt;br/&gt;If you have a strong CPU you can set this value high.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>99999999</number>
      </property>
      <property name="value">
       <number>8</number>
      </property>
     </widget>
    </item>
    <item row="5" column="2">
     <widget class="QPushButton" name="buttonTestArgon">
      <property name="text">
       <string>Test</string>
      </property>
     </widget>
    </item>
    <item row="5" column="0" colspan="2">
     <widget class="QProgressBar" name="progressBar">
      <property name="maximum">
       <number>5000</number>
      </property>
      <property name="value">
       <number>0</number>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <property name="format">
       <string>%v ms</string>
      </property>
     </widget>
    </item>
    <item row="4" column="0" colspan="3">
     <widget class="QLabel" name="label_8">
      <property name="text">
       <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;justify&quot;&gt;Necessary time to calculate your encryption key depending on the taken settings. More calculation time is safer against dictionary and brute force attacks. On the other hand it will also take more time to open your password container. A good value is between &lt;span style=&quot; font-weight:600;&quot;&gt;500ms and 2000ms&lt;/span&gt;. This time will decrease if you have got a more powerful CPU.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
      <property name="textFormat">
       <enum>Qt::RichText</enum>
      </property>
      <property name="scaledContents">
       <bool>false</bool>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item row="3" column="0" colspan="3">
     <widget class="Line" name="line">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <tabstops>
  <tabstop>textOldPassword</tabstop>
  <tabstop>textPassword</tabstop>
  <tabstop>textRepeatedPassword</tabstop>
  <tabstop>buttonSelectFile</tabstop>
  <tabstop>textFileName</tabstop>
  <tabstop>spinLanes</tabstop>
  <tabstop>spinIterationCost</tabstop>
  <tabstop>buttonTestArgon</tabstop>
  <tabstop>spinMemoryCost</tabstop>
 </tabstops>
 <resources>
  <include location="../../resources/resource.qrc"/>
 </resources>
 <connections/>
</ui>
